common/buildcraft/transport/statements/TriggerPowerRequested.java

```diff
@@ -0,0 +1,39 @@
+package buildcraft.transport.statements;
+
+import javax.annotation.Nullable;
+
+import buildcraft.api.core.render.ISprite;
+import buildcraft.api.statements.IStatementContainer;
+import buildcraft.api.statements.IStatementParameter;
+import buildcraft.api.statements.ITriggerInternal;
+import buildcraft.api.transport.pipe.IPipeHolder;
+import buildcraft.core.statements.BCStatement;
+import buildcraft.lib.misc.LocaleUtil;
+import buildcraft.transport.BCTransportSprites;
+import buildcraft.transport.pipe.flow.PipeFlowPower;
+
+public class TriggerPowerRequested extends BCStatement implements ITriggerInternal {
+
+    public TriggerPowerRequested() {
+        super("buildcraft:powerRequested");
+    }
+
+    @Override
+    public boolean isTriggerActive(IStatementContainer source, IStatementParameter[] parameters) {
+        final PipeFlowPower flow = (PipeFlowPower) ((IPipeHolder) source.getTile()).getPipe().getFlow();
```

Comment:
Sorry about my last bit of advice, clearly I had forgotten how triggers worked :/

```suggestion
        if (!(source instanceof IGate)) {
            return false;
        }
        IPipeFlow f = ((IGate) source).getPipeHolder().getPipe().getFlow();
        if (!(f instanceof PipeFlowPower)) {
            return false;
        }
        final PipeFlowPower flow = (PipeFlowPower) f;
```